File: javaNode/src/main/java/org/kevoree/library/compare/KevoreeThreadGroup.java
package org.kevoree.library.compare;

import org.kevoree.log.Log;

/**
 *
 * Created by duke on 9/26/14.
 */
public class KevoreeThreadGroup extends ThreadGroup {

    private String instancePath;

    public KevoreeThreadGroup(String instancePath) {
        super("kev_instance_" + instancePath);
        this.instancePath = instancePath;
    }

    public void uncaughtException(Thread t, Throwable e) {
        Log.error("Uncaught exception in Thread: {}", e, instancePath);
    }
}

File: java/org.kevoree.library.java.javaNode/src/main/java/org/kevoree/library/defaultNodeTypes/KevoreeThreadGroup.kt
package org.kevoree.library.defaultNodeTypes

import org.kevoree.log.Log

/**
 * Created by duke on 8/7/14.
 */

class KevoreeThreadGroup(val n: String) : ThreadGroup(n) {

    override fun uncaughtException(t: Thread, e: Throwable) {
        Log.error("Uncatched exception into Kevoree component {}", e, n);
        super<ThreadGroup>.uncaughtException(t, e)
    }


}